sdk/storage/azure-storage-blob-batch/src/test/java/com/azure/storage/blob/batch/BuilderTest.groovy

```diff
@@ -0,0 +1,103 @@
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.batch
+
+import com.azure.core.http.*
+import com.azure.core.test.http.MockHttpResponse
+import com.azure.core.util.CoreUtils
+import com.azure.storage.blob.BlobContainerClientBuilder
+import com.azure.storage.blob.BlobServiceClientBuilder
+import com.azure.storage.blob.implementation.util.BlobUserAgentModificationPolicy
+import com.azure.storage.common.StorageSharedKeyCredential
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
+import reactor.test.StepVerifier
+import spock.lang.Specification
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+class BuilderTest extends Specification {
+
+    static def credentials = new StorageSharedKeyCredential("accountName", "accountKey")
+    static def endpoint = "https://account.blob.core.windows.net/"
+    private static final Map<String, String> PROPERTIES =
+        CoreUtils.getProperties("azure-storage-blob-batch.properties");
+    private static final String SDK_NAME = "name";
+    private static final String SDK_VERSION = "version";
+    private static String clientName = PROPERTIES.getOrDefault(SDK_NAME, "UnknownName");
+    private static String clientVersion = PROPERTIES.getOrDefault(SDK_VERSION, "UnknownVersion");
+
+    static HttpRequest request(String url) {
+        return new HttpRequest(HttpMethod.HEAD, new URL(url), new HttpHeaders().put("Content-Length", "0"),
+            Flux.empty())
+    }
+
+    def "Construct from service client BlobUserAgentModificationPolicy"() {
+        setup:
+        def serviceClient = new BlobServiceClientBuilder()
+            .endpoint(endpoint)
+            .credential(credentials)
+            .httpClient(new UAStringTestClient("azsdk-java-azure-storage-blob/\\d+\\.\\d+\\.\\d+[-beta\\.\\d+]* azsdk-java-" + clientName + "/" + clientVersion + " " + "(.)*"))
+            .buildClient()
+
+        when:
+        def batchClient = new BlobBatchClientBuilder(serviceClient).buildClient()
+        def pipeline = batchClient.client.client.getHttpPipeline()
+        def foundPolicy = false
+        for (int i = 0; i < pipeline.getPolicyCount(); i++)
+            foundPolicy |= (pipeline.getPolicy(i) instanceof BlobUserAgentModificationPolicy)
+
+
+        then:
+        foundPolicy
+        StepVerifier.create(pipeline.send(request(serviceClient.getAccountUrl())))
+            .assertNext({ it.getStatusCode() == 200 })
+            .verifyComplete()
+    }
+
+    def "Construct from container client BlobUserAgentModificationPolicy"() {
+        setup:
+        def containerClient = new BlobContainerClientBuilder()
+            .endpoint(endpoint)
+            .containerName("containerName")
+            .credential(credentials)
+            .httpClient(new UAStringTestClient("azsdk-java-azure-storage-blob/\\d+\\.\\d+\\.\\d+[-beta\\.\\d+]* azsdk-java-" + clientName + "/" + clientVersion + " " + "(.)*"))
```

Comment:
Can you document some of this and about what exactly it's supposed to be matching